File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.js
var lengthOfLongestSubstring = function(s) {
    let max = 0;
    let start = 0;
    const seen = {};
    for (let i = 0;i < s.length;i++){
        let char = s[i];
        if (char in seen && seen[char]+1 > start){
            start = seen[char]+1;
        }
        seen[char] = i;
        max = Math.max(max, i+1-start);
    }
    return max;
    
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
      seen = {}
      maxi = 0
      start = 0
      for i in range(len(s)):
        char = s[i]
        if char in seen:
            start = max(start, seen[char]+1)
        seen[char] = i
        maxi = max(maxi, i+1 - start)
      return maxi